src/alexandria3k/uspto_zip_cache.py

```diff
@@ -32,15 +32,29 @@ class UsptoZipCache:
 
     def __init__(self):
         self.cached_path = None
-        self.cached_data = None
+        self.cached_data = []
         self.file_name = None
 
-    def read(self, zip_path):
-        """Return a list of XML files in the specified zip file"""
+    def read(self, zip_path, xml_sampling=lambda x: (x[0], x[1])):
+        """Return a list of XML containers in the specified zip file.
+        Sampling is a callable that returns a tuple of two values.
+        When the tuple's second value(x[1]) is `True` the  container
+        will get processed, when it is `False` the container will
+        get skipped. Repeated Zip file decompression is avoided through
+        caching.
 
+        :param zip_path: Path to the Zip file.
+
+        :param xml_sampling: A callable that returns a tuple of two values
+        to control container sampling, defaults to:
+        `lambda x: (x[0], x[1])`, with both x[0] and x[1] being True.
```

Comment:
Please fix this as well, and review the rest of the file for other needed adjustments.